### PR TITLE
Stage timeline refactor

### LINKED ISSUE #93: Stage timeline improvements
- the timeline resizes annoyingly when zooming
- we should display the stage names all the time, not just on mouse-over
- we should indicate when the exit status is not zero
- a tooltip showing start/end times, exit status, etc. would be useful

### PR DESCRIPTION
**Pull request by mwhamgenomics**
- stages are now coloured depending on their exit status - zero -> green, nonzero -> red, not set -> blue
- non-completed stages now extend from start time to current time
- added a tooltip describing exact start/end times and exit statuses
- updated vis to 4.19.1, moment to 2.18.1

Closes #93.

### CODE REVIEW

File: reporting_app/static/stage_timeline.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        t += 'Finished: ' + _reformat(stage.date_finished) + '<br/>';
+        t += 'Exit status: ' + stage.exit_status + '<br/>';
+    }
+    t += '</div>';
+    return t;
+}
+
+
+function stage_timeline(div_id, proc) {
+    var stages = proc.stages.map(
```
Comment (tcezard) [inline reporting_app/static/stage_timeline.js]: Should test that you have stages here as some analysis_driver_proc do not have any

File: reporting_app/templates/proc_report.html [hunk 2]
```diff
@@ -1,91 +1,75 @@
...
-        var items = new vis.DataSet(stage_names);
-        startDate = (moment('{{ proc._created }}', "DD-MM-YYYY HH:mm:SS").format('YYYY-MM-DD HH:mm:SS'))
-        startMargin = moment(startDate).subtract(1, 'days')
-        endMargin = moment(startDate).add(6, 'days')
-        var options = {
-            min: startMargin,
-            max: endMargin
-        }
-        var timeline = new vis.Timeline(container, items, options);
+        stage_timeline("timeline_{{ proc.proc_id }}", {{ proc|safe }});
```
Comment (tcezard) [inline reporting_app/templates/proc_report.html]: It seems to work with just `safe` but I would pass it through `tojson` as well to make sure that the conversion is properly done

File: reporting_app/templates/proc_report.html [hunk 3]
```diff
@@ -1,91 +1,75 @@
...
-        {% if procs %}
-            <ul class="nav nav-tabs" id="analysis_driver_proc_ul">
-                <li class="nav active"><a data-target="#analysis_driver_proc_1_tab" data-toggle="tab">{{ procs[0]._created }}</a></li>
-                {% for proc in procs[1:] %}
-                    <li class="nav"><a data-target="#analysis_driver_proc_{{ loop.index + 1 }}_tab" data-toggle="tab">{{ proc._created }}</a></li>
-                {% endfor %}
-            </ul>
-
-            <div id="analysis_driver_proc_1_tab" class="tab-pane fade in active"> {{ proc_timeline(procs[0]) }} </div>
+<h2>Analysis Driver processes</h2>
```
Comment (tcezard) [inline reporting_app/templates/proc_report.html]: This removed the collapsable title. Was that intended?

Comment (mwhamgenomics) [inline reporting_app/templates/proc_report.html]: That was not intended - I wanted to keep it collapsible, but show it by default when the page loads. Turns out removing the 'collapse' from the `analysis_driver_proc_tabs` div does it.

**Pull request merged**